drivers/storage/vbox/storage/vbox_storage.go

```diff
@@ -674,7 +675,7 @@ func (d *driver) attachVolume(
 		return err
 	}
 
-	if err := m.Refresh(); err != nil {
+	if err = m.Refresh(); err != nil {
```

Comment:
Hi @codenrhoden,

While I agree about scoping, the linter is concerned with [shadowing](https://golang.org/doc/go1#shadowing) -- the unintentional process of referring to a different variable of the same name.

For example:

```go
func (r *router) volumes(
	ctx types.Context,
	w http.ResponseWriter,
	req *http.Request,
	store types.Store) error {

	filter, err := parseFilter(store)
	if err != nil {
		return err
	}
	if filter != nil {
		store.Set("filter", filter)
	}

	var (
		tasks   = map[string]*types.Task{}
		taskIDs []int
		opts    = &types.VolumesOpts{
			Attachments: store.GetAttachments(),
			Opts:        store,
		}
		reply = types.ServiceVolumeMap{}
	)

	for service := range services.StorageServices(ctx) {

		run := func(
			ctx types.Context,
			svc types.StorageService) (interface{}, error) {

			ctx = context.WithStorageService(ctx, svc)

			var err error
			if ctx, err = context.WithStorageSession(ctx); err != nil {
				return nil, err
			}

			return getFilteredVolumes(ctx, req, store, svc, opts, filter)
		}

		task := service.TaskExecute(ctx, run, schema.VolumeMapSchema)
		taskIDs = append(taskIDs, task.ID)
		tasks[service.Name()] = task
	}
```

In the above code snippet from [`api/server/router/volume/volume_routes.go#L20-L63`](https://github.com/codedellemc/libstorage/blob/master/api/server/router/volume/volume_routes.go#L20-L63), the `ctx` variable in the lambda `run` shadows the `ctx` variable from the outer function `volumes`. I do this on purpose as there is no harm and I prefer the consistency of the same variable name. However, someone may, unaware, mean to refer to the `ctx` from the parent `volumes` function from inside the `run` lambda, and it will not work.